### ISSUE TITLE
Part: boolean operation fails with wire-offset-extruded solid

### ISSUE THREAD

**Issue by steel-ne** [FreeCAD/FreeCAD#18047]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Steps to reproduce:
- Part: construct solid with a hole (sketch+extrude)
- Part: select face with hole, click "Create shape element copy"
- Draft: Downgrade copied face to wires
- Part: Select wire of hole, apply 2D offset -2mm (inside a hole), select "Fill offset"
- Part: Extrude offset to 3mm forward and 10 mm reverse direction
- Part: Create simple shape box, allocate it to intersect with the extrude
- Try any boolean (cut, union, etc), found error in a console: "Cut: A fatal error occurred when running boolean operation"
![image](https://github.com/user-attachments/assets/d1508824-3169-4a67-9b01-dfdfaa67899f)
![image](https://github.com/user-attachments/assets/aa3edece-6aff-4075-b3e8-a195f297a62f)
![image](https://github.com/user-attachments/assets/80b53647-3874-427d-85c3-3e71fcb0f431)

### Full version info

```shell
OS: Ubuntu 22.04.5 LTS (ubuntu:GNOME/ubuntu/xcb)
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda AppImage
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: unset/unset/Qt default
Installed mods: 
  * Assembly4 0.50.13
  * Curves 0.6.39
```

### Subproject(s) affected?

None

### Anything else?

File to demonstrate a problem. Open and try to cut Box and Extrude001
[boolean-problem.zip](https://github.com/user-attachments/files/17850134/boolean-problem.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tarman3** [comment]
Offset2D with Fill=true is broken

**Comment by steel-ne** [comment]
Definitely. If I workaround it with "Offset2D" -> "Part_MakeFace" problem is gone.

**Comment by steel-ne** [comment]
I can not find an issue that Offset2D with Fill is broken, leave it open

**Comment by tarman3** [comment]
I think the problem is a combination of conditions:
- Create shape element copy
- Offset2D with Fill

**Comment by CoreyBurger** [comment]
I can still replicate this with FreeCAD 1.0.2 on Windows (I was the original filer of #19054)

**Comment by Fmstrat** [comment]
I just replicated this in Linux as well. I can supply files if needed.

Edit: Actually, my issue may be different as the cut isn't failing, it's just not doing anything. It occurs when tying to cut an primitive from a solid generated from an STL mesh.

**Comment by FEA-eng** [comment]
I can't reproduce the issue with the attached file

```
OS: Windows 10 build 19045
Architecture: x86_64
Version: 26.3.0dev.20260728 (Git shallow)
Build date: 2026/07/28 21:59:12
Build type: Release
Branch: (HEAD detached at 63af5cb2f)
Hash: 63af5cb2f3dbb2b4c7333f6d220b03e05bf92f99
Python 3.11.14, Qt 6.8.3, Coin 4.0.10 (bundled), Pivy 0.6.11 (bundled), Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, Clipper2 , IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL) [ OS: Polish/Poland (pl_PL) ]
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
QPA/File dialog/Color dialog: windows/via Win32/via Qt
Navigation Style/Orbit Style/Rotation Mode: Blender/Rounded Arcball/Window center
Logical DPI/Physical DPI/Pixel Ratio: 96/90.0633/1
OpenGL version/vendor/renderer: 4.6.0 NVIDIA 474.82/NVIDIA Corporation/Quadro K4000/PCIe/SSE2
```

**Comment by CoreyBurger** [comment]
I was able to replicate this in FreeCAD 1.1.3. I used the 6x6 from OpenLock tessellation templates and tried to cut a 5"x5" part from it. See the attached two screenshots

<img width="1338" height="715" alt="Image" src="https://github.com/user-attachments/assets/89cf84ee-6d83-4727-99e7-b145b1d9bfd0" />

<img width="1175" height="930" alt="Image" src="https://github.com/user-attachments/assets/55b0f91c-9e9c-4d73-b2ef-ddcbf56344d6" />

FreeCAD details
OS: Windows 11 build 26200
Architecture: x86_64
Version: 1.1.3.20260725 (Git shallow)
Build date: 2026/07/25 04:52:02
Build type: Release
Branch: grafted,grafted
Hash: 145529fe741292ff0b3977a01195bf0247425794
Python 3.11.14, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.4, OCC 7.8.1
Locale: English/Canada (en_CA)
Navigation Style/Orbit Style/Rotation Mode: CAD/Trackball/Drag at cursor
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/
Logical DPI/Physical DPI/Pixel Ratio: 96/101.6/1
Installed mods: 
  * Curves workbench 0.6.70
